File: Chapter 04/JS/section_4_3_cross_cutting_concerns/before/staticFileLogger.js
/**
 * ANTI-PATTERN: GLOBAL LOGGING UTILITY.
 * ARCHITECTURE NOTE: Exporting a singleton or static method 
 * creates a rigid dependency for any module that imports it.
 */
class StaticFileLogger {
    static log(message) {
        console.log(`(BEFORE_LOGGER) Static Log: ${message}`);
    }
}

module.exports = StaticFileLogger;

File: Chapter 04/Python/section_4_3_cross_cutting_concerns/before/static_file_logger.py
"""
ANTI-PATTERN: MODULE-LEVEL STATIC LOGGING.
ARCHITECTURE NOTE: In Python, module-level functions often act 
as global state. While easy to use, they make unit testing 
difficult because they cannot be easily mocked or replaced.
"""
class StaticFileLogger:
    @staticmethod
    def log(message: str):
        print(f"(BEFORE_LOGGER) Static Log: {message}")